authentication/views.py

```diff
@@ -104,44 +114,89 @@ def post(self, request):
-            return Response({'error': 'Bad request'}, status=status.HTTP_400_BAD_REQUEST)
+            return Response({'error': Error.USER_NOT_FOUND.msg}, status=Error.USER_NOT_FOUND.status)
 
-        jwt_token = Utils.generate_token(user.email, user.pk)
-        reset_link = f"{settings.FRONTEND_URL}/auth/password-reset/{jwt_token}/"
-        try:
-            Utils.send_password_reset_email.delay(email, reset_link)
-        except Exception as e:
-            return Response({'error': 'Failed to send email'}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        access_token = TokenManager.generate_token_for_user(user)
+        reset_link = f"{settings.FRONTEND_URL}/auth/password-reset/{access_token}/"
+
+        Utils.send_password_reset_email(email, reset_link)  # TO DO: REWRITE WITH DECORATORS
         return Response({'message': 'Password reset email sent successfully'}, status=status.HTTP_200_OK)
 
 
 class PasswordResetView(APIView):
-    # This view will be rewritten after implementing custom authentication into the main branch.
+    """
+        A view for handling password reset requests.
+
+        This view allows users to reset their password by providing a valid JWT token and a new password.
+        Upon receiving a valid request, it resets the user's password and sends a password update email.
+
+        Note:
+            This view does not require authentication or permissions.
+
+        Request:
+            - Method: POST
+            - URL: /auth/password-reset/<jwt_token>/
+            - Data Params:
+                - password: The new password for the user (required)
+            - Response:
+                - 200 OK: Password reset successfully
+                - 400 Bad Request: Invalid token or password format
+    """
+    authentication_classes = ()
+    permission_classes = ()
+
     serializer_class = PasswordRecoverySerializer
 
     def post(self, request, jwt_token):
-        uid, email, exp = Utils.decode_token(jwt_token)
-        if uid and email:
-            user = Utils.get_user(uid, email)
-            if user is not None:
-                serializer = self.serializer_class(instance=user, data=request.data)
-                try:
-                    serializer.is_valid(raise_exception=True)
-                    serializer.save()
-                    return Response({'message': 'Password reset successfully'}, status=status.HTTP_200_OK)
-                except ValidationError as e:
-                    return Response({'error': e.message}, status=status.HTTP_400_BAD_REQUEST)
-            else:
-                return Response({'error': 'Invalid token for password reset'}, status=status.HTTP_400_BAD_REQUEST)
+        """Handle POST requests for password reset"""
+
+        payload = TokenManager.get_payload(jwt_token)
+        user_id = payload.get('user_id')
+        if user_id is None:
+            return Response({'error': Error.INVALID_TOKEN.msg}, status=Error.INVALID_TOKEN.status)
```

Comment:
It's good that you're checking if `user_id` is `None`. But it looks like your `TokenManager.get_payload()` method doesn't include robust error handling for cases where the token is invalid, expired, or tampered with. Consider also handling other potential issues with the token, such as signature verification failures

Reply:
The fact of the matter if that SimpleJWT returns 1 exception for all errors (I include screenshots). The TokenError inherits from Python Exception.
![image](https://github.com/Project-Stage-Academy/UA-1131_Forum-PS/assets/76693557/91b89f9a-73d5-4d59-a3f4-7c3feef7612b)
![image](https://github.com/Project-Stage-Academy/UA-1131_Forum-PS/assets/76693557/50233d74-6b01-41d9-bc00-f5e27897c212)